### ISSUE TITLE
Trying to make an example with Remix.run

### ISSUE THREAD

**Issue by arthuryeti** [terra-money/wallet-provider#72]
Hey there,

I'm trying to create an example with Remix.run. But I'm experiencing issues with important terra.js on browser.

```
PublicKey.ts:12 Uncaught ReferenceError: Buffer is not defined

GET /node_modules/@terra-money/terra.js/src/core/staking/msgs/index.ts 404 - - 25.690 ms
GET /node_modules/@terra-money/terra.js/src/core/PublicKey.ts 404 - - 7.521 ms
GET /node_modules/@terra-money/terra.js/src/index.ts 404 - - 8.063 ms
```

Any idea?

**Comment by iamssen** [comment]
@arthuryeti https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler.ts#L4

It seems that remix-run is using esbuild.

It's a similar situation to vite, so refer to these examples and try.

You should create objects such as window.Buffer before the client side app starts.

https://github.com/terra-money/wallet-provider/blob/main/templates/vite/src/index.tsx

**Comment by arthuryeti** [comment]
Alright let me dig deeper. Remix has a guard when you want to import builtin node module.

I'll update this once I find something.

**Comment by RomainLanz** [comment]
How are you going to use the extension on the server side?
I don't believe Remix is a tool adapted for that kind of application at all.

**Comment by tk-o** [comment]
Could you simply skip rendering the WalletProvider component entirely on the server? It's only useful in the client, really.

Unless I'm missing some use case, happy to be corrected ;)

**Comment by RomainLanz** [comment]
The extension also gives you the endpoint to use to query and post data and the connected wallet address.

Even if you pass that information to the server somehow, you need to refresh the page if people make changes in the extension, which is not a very good UX.

**Comment by tk-o** [comment]
Please help understand the use case. I'll write what I think you meant: there's some information that the backend service would need to query from the blockchain, and later display on the client, right?

What about the need for refreshing the page? Why would it be required?

**Comment by tk-o** [comment]
I created [an example Remix app](https://remix-etqr7d--3000.local.webcontainer.io/), that allows for a component to be rendered on the client-side only. 
[Here's the source code](https://stackblitz.com/edit/remix-etqr7d?file=app%2Fcomponents%2Fclient-only.tsx).

![image](https://user-images.githubusercontent.com/3687215/150942657-6af95e99-08bb-4d89-9c0b-37d2556b9e7d.png)

**Comment by RomainLanz** [comment]
I am not sure you understand why Remix has been created and when you should use it.
The whole goal of Remix is to help do SSR and progressive enhancement by fetching and rendering on the server.

In the case of Terra Ecosystem, nearly all interactions need to have information from a client-side extension (Terra Station). Like the connected wallet, LCD endpoints, etc.

If you are going to remove all server-side rendering, why bother to use Remix first?

**Comment by tk-o** [comment]
Sir, you can query blockchain on the server-side, cache the responses, and render the shell of the app with listing, so the load times could improve.

Signing and sending transactions require wallet access, querying Terra Blockchain (LCD endpoints) can be done either on the server-side, or the client-side.

What do you think about it, @RomainLanz?

**Comment by tk-o** [comment]
Correct, changing the network would require querying new data. Also, I agree knowing wallet address and the LCD endpoint to query is not possible on the initial page load.

However, you might store the user's wallet address/LCD endpoint in a cookie/a request header, and have it delivered to the server for whatever you might need it there.

Thanks @RomainLanz for good discussion, it's helping to build more context for others reading this issue later 🤝 

Perhaps we'd need to talk about a specific use-case @arthuryeti had on his mind.

**Comment by arthuryeti** [comment]
Remix.run is so much fun to use, and if it can solve the query part of our dapp. Why not trying?

I've finally manage to have a starter template that is working. @iamssen should I create a PR?

**Comment by iamssen** [comment]
> I've finally manage to have a starter template that is working. @iamssen should I create a PR?

PR is always welcome~
